### ISSUE TITLE
Infinite loop in deflate::compress_to_vec() on empty or one-byte slices

### ISSUE THREAD

**Issue by Vlad-Shcherbina** [Frommi/miniz_oxide#75]
To reproduce, run the following program (with `miniz_oxide 0.3.6`):
```rust
fn main() {
    miniz_oxide::deflate::compress_to_vec(&[], 6);
}
```
Expected result: it should terminate instantly.
Actual result: it does not terminate.

-----

Same for
```rust
fn main() {
    miniz_oxide::deflate::compress_to_vec(&[0], 6);
}
```

**Comment by oyvindln** [comment]
Ah somehow missed this issue being posted, will look into it.

**Comment by Lokathor** [comment]
Hey I just also hit this bug. Can we get a minor release to the crate so the fix is on crates.io?

**Comment by oyvindln** [comment]
Yeah, was hoping to get the fix for #74 as well, but haven't got confirmation on it yet.
